Tests/ArcGISToolkitTests/PreplannedMapModelTests.swift

```diff
@@ -279,18 +279,61 @@ class PreplannedMapModelTests: XCTestCase {
         let area = try XCTUnwrap(areas.first)
         let areaID = try XCTUnwrap(area.portalItem.id)
         
+        let model = PreplannedMapModel(
+            offlineMapTask: task,
+            mapArea: area,
+            portalItemID: portalItem.id!,
+            preplannedMapAreaID: areaID,
+            // User notifications in unit tests are not supported, must pass false here
+            // or the test process will crash.
+            showsUserNotificationOnCompletion: false
+        )
+        
         defer {
             // Clean up JobManager.
             OfflineManager.shared.jobManager.jobs.removeAll()
             
             // Clean up folder.
-            let directory = FileManager.default.preplannedDirectory(
-                forPortalItemID: portalItem.id!,
-                preplannedMapAreaID: areaID
-            )
-            try? FileManager.default.removeItem(at: directory)
+            model.removeDownloadedPreplannedMapArea()
         }
         
+        var statuses: [PreplannedMapModel.Status] = []
+        var subscriptions = Set<AnyCancellable>()
+        model.$status
+            .receive(on: DispatchQueue.main)
+            .sink { statuses.append($0) }
+            .store(in: &subscriptions)
+        
+        await model.load()
+        
+        // Start downloading.
+        await model.downloadPreplannedMapArea()
+        
+        // Wait for job to finish.
+        _ = await model.job?.result
+        
+        // Verify that mobile map package can be loaded.
+        let map = await model.loadMobileMapPackage()
+        XCTAssertNotNil(map)
+        
+        // Give the final status some time to be updated.
+        try? await Task.sleep(nanoseconds: 1_000_000)
+        
+        // Verify statuses.
+        XCTAssertEqual(
+            statuses,
+            [.notLoaded, .loading, .packaged, .downloading, .downloading, .downloaded]
+        )
+    }
+    
+    @MainActor
+    func testRemoveDownloadedPreplannedMapArea() async throws {
+        let portalItem = PortalItem(portal: Portal.arcGISOnline(connection: .anonymous), id: .init("acc027394bc84c2fb04d1ed317aac674")!)
```

Comment:
This line is rather long.

```suggestion
        let portalItem = PortalItem(
            portal: .arcGISOnline(connection: .anonymous),
            id: .init("acc027394bc84c2fb04d1ed317aac674")!
        )
```